dataflow/src/main/java/org/checkerframework/dataflow/reachingdefinitions/ReachingDefinitionsStore.java

```diff
@@ -0,0 +1,116 @@
+package org.checkerframework.dataflow.reachingdefinitions;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.Store;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.cfg.visualize.CFGVisualizer;
+import org.checkerframework.dataflow.expression.JavaExpression;
+import org.checkerframework.javacutil.BugInCF;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.StringJoiner;
+
+/** A reaching definitions store contains a set of reaching definitions represented by nodes. */
+public class ReachingDefinitionsStore implements Store<ReachingDefinitionsStore> {
+
+    /** A set of reaching definitions abstract values. */
+    private final Set<ReachingDefinitionsValue> reachingDefSet;
+
+    /** Create a new ReachDefinitionStore. */
+    public ReachingDefinitionsStore() {
+        reachingDefSet = new LinkedHashSet<>();
+    }
+
+    /**
+     * Create a new ReachDefinitionStore.
+     *
+     * @param reachingDefSet a set of reaching definitions abstract values
+     */
+    public ReachingDefinitionsStore(Set<ReachingDefinitionsValue> reachingDefSet) {
+        this.reachingDefSet = reachingDefSet;
+    }
+
+    /**
+     * Remove the information of a reaching definition from the reaching definitions set.
+     *
+     * @param defTarget target of a reaching definition
+     */
+    public void killDef(Node defTarget) {
+        Iterator<ReachingDefinitionsValue> it = reachingDefSet.iterator();
+        while (it.hasNext()) {
+            ReachingDefinitionsValue existedDef = it.next();
+            if (existedDef.def.getTarget().toString().equals(defTarget.toString())) {
```

Comment:
Could you check the implementation of `equals` for nodes? Maybe you don't need `toString` when `equals` works properly for nodes.

Reply:
I think it works well, let me double check it.

Reply:
You should use reference equality for nodes.

Reply:
Thanks, done.